libmattermost.c

```diff
@@ -3639,12 +3638,14 @@ mm_got_history_of_room(MattermostAccount *ma, JsonNode *node, gpointer user_data
 
 	gint i, len = json_array_get_length(order);
 
+	gint64 since = mm_get_channel_approximate_view_time(ma, channel);
+	since = since - MATTERMOST_DAYS_UPTO_POSTS_FETCHED_IN_UNIX_TIME;
+
 	// do not show updates (such as reactions). only edits, new posts and deletes
+	// also, delete the posts which are before since

```

Comment:
this was already done on line `3649` where a variable `since` is considered to remove post using `json_array_remove_element` we have just changed the definition/value of `since` on line `3642`